lib/kafka/consumer.rb

```diff
@@ -263,7 +263,7 @@ def each_batch(min_bytes: 1, max_wait_time: 5)
             mark_message_as_processed(batch.messages.last)
           end
 
-          @offset_manager.commit_offsets_if_necessary
+          @offset_manager.commit_offsets_if_necessary if @offset_manager.offset_commit_enabled
```

Comment:
Another option is to update `#commit_offsets_if_necessary` to do nothing when `offset_commit_enabled` is false. And i think it wouldn't be a bad option because the `#commit_offsets` is exposed on consumer. What do you think?